src/hooks/useValue/useValue.ts

```diff
@@ -0,0 +1,14 @@
+import { useRef } from 'react';
+
+/**
+ * Keep track of a value that is initialized once and then never changes.
+ *
+ * This behavior is similar to `useState`, but the value is never updated.
+ */
+export function useValue<T>(initializeFunction: () => T): T {
+  const valueRef = useRef<T | null>(null);
+  if (!valueRef.current) {
```

Comment:
So if the initial value is falsy the value can still change? I think that shouldn't happen right?

I'd use the useState hook in the implementation.

```ts
const [value] = useState(initalizeFunction):
return value;
```